lib/Dialect/LLHD/Simulator/State.cpp

```diff
@@ -156,33 +242,42 @@ Slot State::popQueue() {
   return pop;
 }
 
-void State::pushQueue(Time t, int index, int bitOffset, APInt &bytes) {
-  Time newTime = time + t;
-  queue.insertOrUpdate(newTime, index, bitOffset, bytes);
-}
-void State::pushQueue(Time t, std::string inst) {
+void State::pushQueue(Time t, unsigned inst) {
   Time newTime = time + t;
   queue.insertOrUpdate(newTime, inst);
   instances[inst].expectedWakeup = newTime;
 }
 
+llvm::SmallVectorTemplateCommon<Instance>::iterator
+State::getInstanceIterator(std::string instName) {
+  auto it =
+      std::find_if(instances.begin(), instances.end(),
+                   [&](const auto &inst) { return instName == inst.name; });
+  if (it == instances.end()) {
+    llvm::errs() << "could not find an instance named " << instName << "\n";
+    exit(EXIT_FAILURE);
```

Comment:
This is generally frowned upon in a library.  Should this be an assertion?  If not, it should probably be explicitly propagated up through callers.

Reply:
Yes, this should totally be an assertion.